### PR TITLE
We need to call record.valid?

### PR DESCRIPTION
**Pull request by jasonm23**
To populate `record.errors` we must first call `.valid?`

### CODE REVIEW

**Comment by jasonm23** [comment]
Travis-ci choked on a gem (term-ansicolor)

FYI local run of `rspec` passed 42 examples, 0 fails.

**Comment by Fivell** [comment]
Hello @jasonm23, failed records should already contain error messages (validations is what activerecord-import is doing when providing validate option). I don't see any reason to re-call valid. What are you trying to fix?

**Comment by jasonm23** [comment]
When we were using the gem our error messages were not included in the flash message until we made this patch.

I saw that you have spec coverage of this, and they pass.

So it's obviously a little weird that we had this problem.

I'm happy to run any additional tests, let me know.

**Comment by Fivell** [comment]
>>When we were using the gem our error messages were not included in the flash message

Would be great to see failing test before this change.

**Comment by Fivell** [comment]
I think this can be related to different ```activerecord-import``` version. However still hard to say without    spec

**Comment by jasonm23** [comment]
I'll post an example tomorrow. Thank you

**Comment by jasonm23** [comment]
Hi @Fivell 

We've figured out why the import result spec is passing, while also seeing the error we are experiencing.

The following spec (not currently in the pull-request branch) will fail and illustrate the issue.

```
context 'with invalid data insert on model validation' do
  let(:options) { { validate: true } }

  before do
    Author.create!(name: 'John', last_name: 'Doe')
  end

  it 'should render both successful and failed message' do
    upload_file!(:authors_invalid_model)
    expect(page).to have_content 'Failed to import 1 author'
    expect(page).to have_content 'Last name has already been taken - Doe'
    expect(page).to have_content 'Successfully imported 1 author'
    expect(Author.count).to eq(2)
  end
end
```

The reason the `spec/import_result_spec.rb` is passing, is because we're creating the `Author` objects with `FactoryGirl.create`, this will trigger validations (thanks due to ActiveRecord create.)

This isn't happening when the records are imported, as we can see if we run the spec above.

See the fail output below:

```
  1) import with valid options when importing file with invalid data insert on model validation should render both successful and failed message
     Failure/Error: expect(page).to have_content 'Last name has already been taken - Doe'
       expected to find text "Last name has already been taken - Doe" in "Rails426 Dashboard Authors Comments Posts Admin / Authors / Import Failed to import 1 author: Successfully imported 1 author Please, select file to import File* Powered by Active Admin 1.0.0.pre4"
     # ./spec/import_spec.rb:314:in `block (5 levels) in <top (required)>'
```

I can add this spec to the pr branch shortly, so we can see it pass.

**Comment by Fivell** [comment]
@jasonm23 , thanks for clarification. I used your expectation to check agains multiple activerecord-import version, after checking I  can tell that this bug is not reproducible before activerecord-import v 0.14.0.

**Comment by Fivell** [comment]
I want to report bug on activerecord-import before merging this

**Comment by Fivell** [comment]
https://github.com/zdennis/activerecord-import/issues/331

**Comment by jasonm23** [comment]
Cool, lets see how that goes.  Our project is using a temporary fork.

**Comment by Fivell** [comment]
@jasonm23 , fix is merged to `activerecord-import` master brach.

**Comment by jasonm23** [comment]
Good news !  Thank you.

File: lib/active_admin_import/import_result.rb [hunk 1]
```diff
@@ -32,6 +32,7 @@ def empty?
     def failed_message(options = {})
       limit = options.fetch(:limit, failed.count)
       failed.first(limit).map do |record|
+        record.valid?
```
Comment (Fivell) [inline lib/active_admin_import/import_result.rb]: please try to change dependency of activerecord-import to '~> 0.16.1' instead of this line

**Comment by Fivell** [comment]
@jasonm23 ,  released 0.16.1 , can you modify your PR?

**Comment by jasonm23** [comment]
Shouldn't we just close this PR?

**Comment by jasonm23** [comment]
Sorry, I guess you want to merge in the spec amendments. 

I'll remove the .valid? Shortly.

**Comment by jasonm23** [comment]
Done, only spec amendment remains

**Comment by Fivell** [comment]
@jasonm23 , would  be great if you can squash commits to single one with right description

**Comment by Fivell** [comment]
>>Shouldn't we just close this PR?

@jasonm23 , yes it will be easier to add spec in separate PR. Thanks!

**Pull request closed without merging**